src/core/service.c

```diff
@@ -3303,9 +3353,15 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
 
         if (is_clean_exit(code, status, clean_mode, &s->success_status))
                 f = SERVICE_SUCCESS;
-        else if (code == CLD_EXITED)
-                f = SERVICE_FAILURE_EXIT_CODE;
-        else if (code == CLD_KILLED)
+        else if (code == CLD_EXITED) {
+                /* ExecCondition= calls that exit with (0, 254] should invoke skip-like behavior instead of failing */
+                if (s->state == SERVICE_CONDITION && s->control_pid == pid && status > 0 && status < 255) {
+                        f = SERVICE_SKIP_CONDITION;
+                        dual_timestamp_get(&u->condition_timestamp);
```

Comment:
hmm, i wonder if we really should set the condition timestamp here. i mean, ExecCondition= already has its own timestamp anyway so we don't really need that, and I don't think i would necessarily conflate the "classic" conditions with ExecCondition too much.

There's also the problem that the condition timestamp is generally set whenever we do a condition check, regardless if with a good or bad result. however in this case you only set this only on failure. 

I'd probably just drop this (and the following line)

Reply:
This was a small hack to get the job_log_done_status_message to show the condition check message (https://github.com/systemd/systemd/blob/master/src/core/job.c#L890) rather than the one for `JOB_DONE`. The timestamp was just included for consistency.

Would it make more sense to get the skip flow to return `JOB_SKIPPED` instead of (ab)using `u->condition_result` to return the right message?

Reply:
uh, not sure how i feel about that, but maybe prep a patch to see how that looks